### PR TITLE
fix(runtime): add root scope id to the user provided nested children as classname

### PR DESCRIPTION
**Pull request by yigityuce**
## What is the current behavior?

GitHub Issue Number: #5749

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by christian-bromann — COMMENTED** [review]
There are some places where we don't use the patched `insertBefore` version:

- https://github.com/yigityuce/stencil/blob/f4344293aca323ad38567621d96d6bf78f861717/src/runtime/dom-extras.ts#L356
- https://github.com/yigityuce/stencil/blob/f4344293aca323ad38567621d96d6bf78f861717/src/runtime/vdom/vdom-annotations.ts#L223
- https://github.com/yigityuce/stencil/blob/f4344293aca323ad38567621d96d6bf78f861717/src/runtime/connected-callback.ts#L131

I wonder if we should stay consistent with this patch and apply it in there too.

Otherwise the code lgtm, I was able to reproduce the problem given the reproducible example and verified that the fix actually addresses the problem.

File: src/runtime/vdom/vdom-render.ts [hunk 1]
```diff
@@ -924,6 +911,52 @@ export const nullifyVNodeRefs = (vNode: d.VNode) => {
...
+  const inserted = parent?.insertBefore(newNode, reference);
+
+  if (BUILD.scoped) {
+    setParentScopeIdAsClassName(newNode as d.RenderNode, parent as d.RenderNode);
+  }
+
+  return inserted;
+};
+
+const findParentScopeId = (element: any): any => {
```
[the remark this answers is not in the document]
Comment (yigityuce) [reply src/runtime/vdom/vdom-render.ts:933]: ahh sorry I hate "any", tough I forgot to update typings

[the remark this answers is not in the document]
Comment (yigityuce) [reply src/runtime/vdom/vdom-render.ts]: I updated the missing `insertBefore` patches in other files too and checked the jest and wdio tests and seems everything works fine

**Review by christian-bromann — APPROVED** [review]
LGTM 👍 

I will let someone from the @ionic-team/stencil team take another look at it.

**Review by alicewriteswrongs — APPROVED** [review]

**Pull request merged**